Melody Ikechukwu Emmanuel/codeFaizal.ts

```diff
@@ -0,0 +1,121 @@
+
+class Laptop{
+    keyboard: Keyboard[] = []
+    nic: NetworkInterfaceCard
+    screen: Display 
+    hardDisk: HardDisk
+    powerState: boolean = false
+    bitKind: BitKind
+    os: OS
+    allUpdateLogs: string[] = [];
+
+    constructor(keyboard: Keyboard, bit: BitKind, display: Display, nic: NetworkInterfaceCard, os: OS, harddisk: HardDisk){
+        this.keyboard.push(keyboard)
+        this.bitKind = bit
+        this.screen = display
+        this.nic = nic
+        this.os = os
+        this.hardDisk = harddisk
+    }
+
+    switchOn() {
+        this.powerState = true
+    }
+
+    switchOff(){
+        this.powerState = false
+    }
+    showAllUpdateLogs(){
+        console.log("All updates made: " + this.allUpdateLogs.join("\n"))
+    }
+
+    update(thingToUpdate: string, newValue: any){

```

Comment:
You didn't do the update assignment.
Meet 'Elochukwu' in the class to explain this